template-tanstack-start/app/router.tsx

```diff
@@ -27,6 +27,7 @@ export function createRouter() {
       routeTree,
       defaultPreload: "intent",
       context: { queryClient },
+      scrollRestoration: true,
```

Comment:
I can confirm this is the behavior that the default TanStack Router template uses now. Let’s do it!